src/payment/strategies/chasepay/chasepay-payment-strategy.ts

```diff
@@ -0,0 +1,275 @@
+    private _methodId?: string;
+    private _paymentMethod?: PaymentMethod;
+    private _storeLanguage?: string;
+
+    constructor(
+        store: CheckoutStore,
+        private _paymentMethodActionCreator: PaymentMethodActionCreator,
+        private _chasePayScriptLoader: ChasePayScriptLoader,
+        private _paymentActionCreator: PaymentActionCreator,
+        private _orderActionCreator: OrderActionCreator,
+        private _requestSender: RequestSender,
+        private _wepayRiskClient: WepayRiskClient,
+        private _paymentStrategyActionCreator: PaymentStrategyActionCreator,
+        private _checkoutActionCreator: CheckoutActionCreator
+
+    ) {
+        super(store);
+    }
+
+    initialize(options: PaymentInitializeOptions): Promise<InternalCheckoutSelectors> {
+        const { methodId } = options;
+
+        if (!options.chasepay) {
+            throw new InvalidArgumentError('Unable to initialize payment because "options.chasepayCheckoutOptions" argument is not provided.');
+        }
+
+        const {
+            onPaymentSelect = () => {},
+            logoContainer,
+            walletButton,
+        } = options.chasepay;
+
+        const storeConfig = this._store.getState().config.getStoreConfig();
+
+        if (!storeConfig) {
+            throw new MissingDataError(MissingDataErrorType.MissingCheckoutConfig);
+        }
+
+        this._storeLanguage = storeConfig.storeProfile.storeLanguage;
+        this._logoContainer = logoContainer;
+        this._methodId = methodId;
+
+        if (options.gatewayId === 'wepay') {
+            this._wepayRiskClient.initialize();
+        }
+
+        this._button = walletButton && document.getElementById(walletButton) || undefined;
+
+        if (this._button) {
+            this._button.addEventListener('click', this._handleWalletButtonClick);
+        }
+
+        return this._setChasePayConfiguration(
+            methodId,
+            onPaymentSelect,
+            this._storeLanguage
+        )
+        .then(() => super.initialize(options));
+    }
+
+    execute(orderRequest: OrderRequestBody, options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        const { payment, ...order } = orderRequest;
+        if (!payment || !options) {
+            throw new PaymentArgumentInvalidError(['payment']);
+        }
+
+        if (!this._paymentMethod || !this._paymentMethod.initializationData) {
+            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+        }
+
+        const { initializationData, config } = this._paymentMethod;
+
+        const riskToken = (payment.gatewayId === 'wepay') ? this._wepayRiskClient.getRiskToken() : undefined;
+        const paymentData = this._mapCardData(initializationData, riskToken);
+
+        return this._getPaymentData(payment.methodId,
+            initializationData.digitalSessionId,
+            config.testMode,
+            this._processPayment(payment, paymentData, order, options))
```

Comment:
The fourth parameter is `onPaymentSelect` - it should be a callback function (the return value of `_processPayment` is not a callback function).

Reply:
Oh, right, got it.